python/ray/serve/_private/http_state.py

```diff
@@ -254,11 +254,11 @@ def __init__(
         else:
             self._config = HTTPOptions()
         self._proxy_states: Dict[NodeId, HTTPProxyState] = dict()
-        self._head_node_id: str = head_node_id
+        self._head_node_id: str = get_head_node_id()
```

Comment:
nit: pass this in from init to avoid duplicating the logic and so unit tests for HTTP state don't need to make ray api calls